apps/src/code-studio/initApp/loadApp.js

```diff
@@ -131,4 +317,85 @@ module.exports = function (callback) {
   } else {
     loadLastAttemptFromSessionStorage();
   }
+}
+
+window.dashboard = window.dashboard || {};
+window.dashboard.project = project;
+
+window.apps = {
+
+  // Set up projects, skipping blockly-specific steps. Designed for use
+  // by levels of type "external".
+  setupProjectsExternal: function () {
+    if (!window.dashboard) {
+      throw new Error('Assume existence of window.dashboard');
+    }
+  },
+
+  // Define blockly/droplet-specific callbacks for projects to access
+  // level source, HTML and headers.
+  // TODO(dave): Extract blockly-specific handler code into _blockly.html.haml.
```

Comment:
This TODO can probably be removed -- our init code could still use some cleanup, but I'm not confident that this specific strategy is the right way to go.